zipline/pipeline/factors/technical.py

```diff
@@ -144,6 +150,238 @@ def compute(self, today, assets, out, close, volume):
+        2017-03-21     .03     .01   -.02
+        2017-03-22     .04    -.02   -.02
+
+    Suppose we are interested in predicting each stock's returns from SPY's
+    over rolling 5-day look back windows. We can compute rolling regression
+    coefficients (alpha and beta) from 2017-03-17 to 2017-03-22 by doing::
+
+        regression_factor = RollingRegressionOfReturns(
+            asset=Equity(24),
+            returns_length=10,
+            regression_length=5,
+        )
+        alpha = regression_factor.alpha
+        beta = regression_factor.beta
+
+    The result of computing ``alpha`` from 2017-03-17 to 2017-03-22 gives::
+
+                       SPY    MSFT     FB
+        2017-03-17       0    .011   .003
+        2017-03-20       0   -.004   .004
+        2017-03-21       0    .007   .006
+        2017-03-22       0    .002   .008
+
+    And the result of computing ``beta`` from 2017-03-17 to 2017-03-22 gives::
+
+                       SPY    MSFT     FB
+        2017-03-17       1      .3   -1.1
+        2017-03-20       1      .2     -1
+        2017-03-21       1     -.3     -1
+        2017-03-22       1     -.3    -.9
+
+    Note that SPY's column for alpha is all 0's and for beta is all 1's, as the
+    regression line of SPY with itself is simply the function y = x.
+
+    To understand how each of the other values were calculated, take for
+    example MSFT's ``alpha`` and ``beta`` values on 2017-03-17 (.011 and .3,
+    respectively). These values are the result of running a linear regression
+    predicting MSFT's returns from SPY's returns, using values starting at
+    2017-03-17 and looking back 5 days. That is, the regression was run with
+    x = [-.03, -.02, -.01, 0, .01] and y = [.03, -.03, .02, -.02, .04], and it
+    produced a slope of .3 and an intercept of .011.
+
+    See Also
+    --------
+    :class:`zipline.pipeline.factors.technical.RollingPearsonOfReturns`
+    :class:`zipline.pipeline.factors.technical.RollingSpearmanOfReturns`
+    """
+    outputs = ['alpha', 'beta', 'r_value', 'p_value', 'stderr']
+    params = ['baseline_asset']
+
+    def __new__(cls,
+                baseline_asset,
+                returns_length,
+                regression_length,
+                mask=NotSpecified,
+                **kwargs):
+        if mask is not NotSpecified:
+            # Make sure we do not filter out the asset of interest.
+            mask = mask | SingleAsset(asset=baseline_asset)
+        return super(RollingRegressionOfReturns, cls).__new__(
+            cls,
+            baseline_asset=baseline_asset,
+            inputs=[Returns(window_length=returns_length)],
+            window_length=regression_length,
+            mask=mask,
+            **kwargs
+        )
+
+    def compute(self, today, assets, out, returns, baseline_asset):
+        asset_col = searchsorted(list(assets), baseline_asset.sid)
+        my_asset = returns[:, asset_col]
+        for i in range(len(out)):
```

Comment:
I couldn't find a good vectorized way of doing this, but I don't think the looping should really be any slower since it would have to compute all the regressions regardless. This is also probably easier to read as it's pretty explicit what its doing.